src/wx/config/strutils.cpp

```diff
@@ -41,40 +60,53 @@ wxArrayString str_split_with_sep(const wxString& text, const wxString& sep)
     return str_split(text, sep, true);
 }
 
-std::vector<uint8_t> utf16_to_utf8(const uint16_t* utf16) {
-    std::vector<uint8_t> out;
-    for (size_t i = 0; utf16[i]; i++) {
-        uint16_t c = utf16[i];
-        if (c < 0x80) {
-            out.push_back(c);
-        } else if (c < 0x800) {
-            out.push_back(0xC0 | (c >> 6));
-            out.push_back(0x80 | (c & 0x3F));
-        } else if (c < 0xD800 || c >= 0xE000) {
-            // Regular 3-byte UTF-8 character.
-            out.push_back(0xE0 | (c >> 12));
-            out.push_back(0x80 | ((c >> 6) & 0x3F));
-            out.push_back(0x80 | (c & 0x3F));
+std::vector<uint8_t> utf16_to_utf8_vector(char16_t* utf16) {
```

Comment:
Make the return type `std::string` instead of `std::vector`, it can do the same things you are doing here and that makes more sense and is easier to use, and will make the code here a bit cleaner.

I see why you said you don't like the iconv API very much, we could have used the simple version of the API without buffer resizing (because we only need small strings) and error handling, but since you already did all of this we may as well keep it!

Reply:
This is a bit pedantic but bear with me. With this function, I want to be able to check the individual code units of the output for testing purposes. While we can use a `std::string` for that purpose, there is some degree of semantic abuse as a string really stores some `char`, which are not necessarily 8-bits and whose signedness is implementation defined (thank you very much, C standard). Furthermore, after double-checking, `char16_t` is not guaranteed to be 16-bits (thank you very much, C++ standard, AGAIN).
This function only really exists for testing purposes and here what we are testing is "are the code units valid?". So since we are exclusively considering code units here, I believe this function should really be taking a `uint16_t*` and return a vector of `uint8_t`.